packages/lodestar/src/chain/chain.ts

```diff
@@ -281,11 +277,11 @@ export class BeaconChain extends (EventEmitter as {new (): ChainEventEmitter}) i
       const listener = (signedBlock: SignedBeaconBlock): void => {
         const root = this.config.types.BeaconBlock.hashTreeRoot(signedBlock.message);
         if (this.config.types.Root.equals(root, blockRoot)) {
-          this.removeListener("processedBlock", listener);
+          this.emitter.removeListener("processedBlock", listener);
           resolve();
         }
       };
-      this.on("processedBlock", listener);
+      this.emitter.on("processedBlock", listener);
     });
   }
```

Comment:
I think this method should also receive abort signal or timeout?
And on chain stop should cancel